lib/tesla/middleware/logger.ex

```diff
@@ -213,8 +212,17 @@ defmodule Tesla.Middleware.Logger do
     |> Enum.map(fn {k, v} -> ["Query: ", to_string(k), ": ", to_string(v), ?\n] end)
   end
 
-  defp debug_headers([]), do: @debug_no_headers
-  defp debug_headers(headers), do: Enum.map(headers, fn {k, v} -> [k, ": ", v, ?\n] end)
+  defp debug_headers([], _opts), do: @debug_no_headers
+
+  defp debug_headers(headers, opts) do
+    filtered = Keyword.get(opts, :filter_headers)
+    filtered = if is_list(filtered), do: filtered, else: []
```

Comment:
I wouldn't do this as it will fail silently when you pass `filter_headers: %{"x" => true}` for example. It's better to just blow the code when something wrong is passed, so I'd keep `Keyword.get(opts, :filter_headers, [])`

Reply:
True. Let it crash.